util/multisig/src/secp256k1.rs

```diff
@@ -0,0 +1,146 @@
+use crate::error::{Error, ErrorKind};
+use log::{debug, trace};
+pub use secp256k1::{
+    All, Error as Secp256k1Error, Message, PublicKey, RecoverableSignature, Secp256k1, SecretKey,
+    Signature,
+};
+
+lazy_static! {
+    pub static ref SECP256K1: Secp256k1<All> = Secp256k1::new();
+}
+
+pub fn sign(message: &Message, sk: &SecretKey) -> Signature {
+    SECP256K1.sign(message, sk)
+}
+
+/// position of each sigs must according to the pk that sined the sig
+/// Example 2 of 3 sigs: [s1, None, s3], pks: [pk1, pk2, pk3]
+pub fn verify_m_of_n(
+    message: &Message,
+    m_threshold: usize,
+    sigs: Vec<Option<Signature>>,
+    pks: Vec<PublicKey>,
+) -> Result<(), Error> {
+    if sigs.len() > pks.len() {
+        Err(ErrorKind::SigCountOverflow)?;
+    }
+    if m_threshold > sigs.len() {
```

Comment:
Need to filter none before checking threshold.

```suggestion
    if m_threshold > sigs.filter_map(|s| s).len() {
```